sql/types/system_bool.go

```diff
@@ -140,9 +140,8 @@ func (t SystemBoolType) Equals(otherType sql.Type) bool {
 }
 
 // MaxTextResponseByteLength implements the Type interface
-func (t SystemBoolType) MaxTextResponseByteLength(_ *sql.Context) uint32 {
-	// system types are not sent directly across the wire
-	return 0
+func (t SystemBoolType) MaxTextResponseByteLength(ctx *sql.Context) uint32 {
+	return t.UnderlyingType().MaxTextResponseByteLength(ctx)
```

Comment:
While debugging the customer issue, I noticed that these system types DO get sent over the wire, so these comments were incorrect. To send the correct field metadata, each system type needs to delegate `MaxTextResponseByteLength` to the underlying type.